### PR TITLE
Completed task 10 - View by topic

### PR DESCRIPTION
**Pull request by Clatherine**
Added functionality to filter by topic
Added error handling in case user tries to type a topic directly into URL

also addressed feedback from task 10:

I did edit to make deletion optimistic, but changed it back since the user experience was a bit odd when there was a failed request - having the comment disappear and then reappear. Hope this is ok. Have added error handling so the user gets a message that the comment could not be deleted if there is a problem

### CODE REVIEW

File: src/Components/ArticleByIdPageComponents/CommentsSectionComponents/CommentCard.jsx [hunk 1]
```diff
@@ -19,19 +20,28 @@ const handleClick=()=>{
...
+        return updatedList
+            })
 })
-return updatedList
-    })
+.catch((err)=>{
+    // setIsError(true)
+    // setTimeout(setIsError, 2500, false)
+    setError(true)
+    setTimeout(setError, 2500, false)
```
Comment (nc-modou) [inline src/Components/ArticleByIdPageComponents/CommentsSectionComponents/CommentCard.jsx:36]: I'd go with isError for a boolean, and error / errorMsg for a string

Also, try to delete commented out code that you're not using anymore - remember that PRs want to be clean and polished, ready to go into main

File: src/api.js [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 
 const articlesApi = axios.create({baseURL: "https://v2-nc-news-backend-project-northcoders.onrender.com"})
 
-export const getArticles = () =>{
-   return articlesApi.get("/api/articles")
+export const getArticles = (topic) =>{
+   return articlesApi.get("/api/articles", {
+      params: {
+         topic: topic
+      }
```
Comment (nc-modou) [inline src/api.js:9]: Great use of axios' ability to handle params!

**Review by nc-modou — COMMENTED** [review]
Very nicely done! Filtering works great and it's cool that it's reflected in the url - this is always good to have as people can bookmark a certain url or share a link to it and so on

Styling wise, I noticed the topic selection bar is really thin, adding a bit to it's height would help with the user experience as it'll me more obvious and easy to click

File: src/Components/ArticleByIdPageComponents/CommentsSectionComponents/CommentCard.jsx (hunk 1 above)
Comment (Clatherine) [reply src/Components/ArticleByIdPageComponents/CommentsSectionComponents/CommentCard.jsx:36]: done!

**Comment by Clatherine** [comment]
Updated styling on topic selection bar

**Pull request merged**